### ISSUE TITLE
"Current status is: Login failed" Error for Github

### ISSUE THREAD

**Issue by JeffSpies** [philipstarkey/bitbucket-hg-exporter#48]
Hi, thanks for building this! Hit a roadblock though:

- I've cleared the key ring.
- BitBucket credentials work fine; the program calls out to hg and downloads all of my repos.
- Then I get the following messages between rounds of sleep:
    Waiting on jeffspies/reponame to complete. Current status is: Login failed
- I'm assuming this is Github, as the BitBucket gathered all the data just fine.
- I'm using a personal access token that I've given access to the full API scope. I've tried it in both password and token (both accepted it), and both have the same results. Cleared the ring, tried again, same behavior.

If there's something I'm missing, let me know! Thanks!

**Comment by scpeters** [comment]
Are you using an automatic import to Github or using your own tool to convert to git? I think if you have private repositories, the automatic import isn't well-supported

**Comment by philipstarkey** [comment]
@scpeters is right. Your BitBucket repositories are private. Using the GitHub importer for that is not supported completely (technically it should work in you uncomment a couple of lines in my code, this is documented in the project readme).

The best approach however would be to change to using a local tool to import (also documented in the readme). You don't have to download everything from BitBucket again, just restart this tool, select your saved config, change the option, and then tell my tool to start. It should quickly skip over the BitBucket download stage and then prompt you to run an external tool to do the conversion. A suggested external tool is on the readme.

**Comment by philipstarkey** [comment]
I should add that if you do want to use GitHub's importer you can probably manually enter the credentials through the GitHub web interface by visiting the new empty repository. It should redirect you to the web import page where you can fix the error. My tool should be fine with this (because it's just monitoring the import status). A bit of a pain though if you have many repositories!

But local tool conversion (and then push to GitHub) is safer because I've found that GitHub drops certain things (multiple heads) and doesn't handle UTF8 characters well when using its own importer. Local tool also doesn't rely on BItBucket being up. If you can see you have the hg repositories and BitBucket API data downloaded from my tool, you should be able to migrate to GitHub even if BitBucket has deleted everything (in theory anyway... obviously I have no way to test this until tomorrow so don't rely on me being right!!)

**Comment by JeffSpies** [comment]
Thanks for the replies. I'll try hg-export-tool now and see if I can get them on GH.

**Comment by philipstarkey** [comment]
You may also need to delete the empty repositories on GitHub that are stuck at the import stage (not 100% sure)

**Comment by JeffSpies** [comment]
Indeed. And they weren't recognized in the API because they weren't initialized yet. :) 

But got it all working after a bit more cleanup of the hg repos.

Thanks again for the tool!

**Comment by philipstarkey** [comment]
Great!  I'll close this for now but feel free to open another issue if you run into further problems. I'll be looking out for issue notifications for the next 8 hours. Depending on which timezone Atlassian is killing the repos, that may be the end of this project!
